tests/checks/read.fish

```diff
@@ -366,3 +366,12 @@ function-scoped-read
 # CHECK: $skamtebord[1]: |bar|
 # CHECK: $craaab: set in local scope, unexported, with 1 elements
 # CHECK: $craaab[1]: |baz|
+
+echo foo\nbar\nbaz | begin
```

Comment:
I would suggest leaving a comment indicating that this is specifically testing chunked vs byte-by-byte reads, just to have pity on whoever needs to track down a problem at a later date :)